File: src/structural/adapter/adapter.py

# Target class
class Target:
    def request(self):
        print("Target: Default behavior")

# Adaptee class
class Adaptee:
    def specific_request(self):
        print("Adaptee: Specific behavior")

# Adapter class
class Adapter(Target):
    def __init__(self, adaptee):
        self._adaptee = adaptee

    def request(self):
        print("Adapter: Adapting...")
        self._adaptee.specific_request()

if __name__ == "__main__":
    target = Adapter(Adaptee())
    target.request()

File: src/structural/adapter/Adapter.cs

using System;

// Target class
public class Target
{
    public virtual void Request()
    {
        Console.WriteLine("Target: Default behavior");
    }
}

// Adaptee class
public class Adaptee
{
    public void SpecificRequest()
    {
        Console.WriteLine("Adaptee: Specific behavior");
    }
}

// Adapter class
public class Adapter : Target
{
    private readonly Adaptee _adaptee;

    public Adapter(Adaptee adaptee)
    {
        _adaptee = adaptee;
    }

    public override void Request()
    {
        Console.WriteLine("Adapter: Adapting...");
        _adaptee.SpecificRequest();
    }
}

// Client code
class Program
{
    static void Main()
    {
        Target target = new Adapter(new Adaptee());
        target.Request();
    }
}
